src/v/redpanda/admin_server.cc

```diff
@@ -1975,12 +1975,31 @@ admin_server::get_broker_handler(std::unique_ptr<ss::httpd::request> req) {
           ss::httpd::reply::status_type::service_unavailable);
     }
 
+    auto h_report
+      = co_await _controller->get_health_monitor().local().get_cluster_health(
+        cluster::cluster_report_filter{
+          .node_report_filter
+          = {.include_partitions = cluster::include_partitions_info::no},
+          .nodes = {id}},
+        cluster::force_refresh::no,
+        model::no_timeout);
+
     ss::httpd::broker_json::broker ret;
     ret.node_id = node_meta->broker.id();
     ret.num_cores = node_meta->broker.properties().cores;
     if (node_meta->broker.rack()) {
         ret.rack = node_meta->broker.rack().value();
     }
+    if (h_report) {
+        const auto& reports = h_report.value().node_reports;
+        auto node_it = std::find_if(
+          reports.begin(), reports.end(), [id](const auto& report) {
+              return report.id == id;
+          });
+        if (node_it != reports.end()) {
+            ret.version = node_it->local_state.redpanda_version;
+        }
+    }
```

Comment:
if the health report isn't available for some reason, will the version still show up in the output json (e.g. as null)?

Reply:
No, the "version" key just doesn't show up. example

```
{
  "node_id": 0,
  "num_cores": 1,
  "membership_status": "active",
  "maintenance_status": {
    "draining": false,
    "finished": false,
    "errors": false,
    "partitions": 0,
    "eligible": 0,
    "transferring": 0,
    "failed": 0
  }
}

```

Reply:
i wonder if it should. like, having a stable schema would be easier to process i would think. i don't have a strong opinion here, maybe ask rpk dev?

Reply:
@r-vasquez you have an opinion on this?

Reply:
No strong opinion either, Go handles missing fields pretty well and we will cover the cases when the version is not there (rpk might be calling older redpanda versions)